rs/execution_environment/src/canister_manager.rs

```diff
@@ -366,8 +366,6 @@ impl CanisterManager {
             canister.system_state.reserved_balance_limit(),
         )?;
 
-        let is_controllers_change = validated_settings.controllers().is_some();
```

Comment:
We have to keep a check if a new canister history entry should be added: now this check would be
```
validated_settings.controllers().is_some() || validated_settings.environment_variables().is_some()
```
and we should only add a canister history entry if this check is satisfied (also if environment variables are enabled). In other words, we should not be adding canister history entries if neither environment variables nor controllers are changed.